example/morphology.cpp

```diff
@@ -0,0 +1,130 @@
+//
+// Copyright 2021 Prathamesh Tagore <prathameshtagore@gmail.com>
+//
+// Use, modification and distribution are subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#include <boost/gil/extension/io/png.hpp>
+#include <boost/gil/image_processing/morphology.hpp>
+#include <vector>
+#include<iostream>
+#include<map>
+#include<string>
+//Default structuring element is SE = [1,1,1]
+//                                    |1,1,1|
+//                                    [1,1,1]
+//SE(1,1)(center pixel) is the one which coincides with the currently considered pixel of the 
+//image to be convolved. The structuring element can be easily changed by the user.
+
+int main(int argc,char **argv)
+{
+    std::map<std::string,bool>operations;
+    if(argc < 4 || argc > 11)
+    {
+        std::cout<<"Wrong format of command line arguments.\n";
+        std::cout<<"Correct format is <Input_image.png> <Output_image_template>"
+        " <operation1> <operation2> <operation3> <operation4> <operation5> <operation6>"
+        " <operation7> <operation8>\n";
+        //User has to enter atleast one operation and they can enter maximum 8 operations
+        //considering binary conversion to be an operation.Output_image_template argument is the 
+        //common component which will be added in all output images followed by a hyphen and the 
+        //operation name.
+        //Example : 
+        //./example_morphology original.png out black_hat top_hat morphological_gradient dilation
+        // erosion opening closing
+        //Order of arguments entered will not matter with the exception of binary operation used for 
+        //binary morphological operations.If binary is entered through the command line,
+        //it will always be the first operation to be applied.
+        return -1;
+    }
+    else
+    {
+        for(int i=3;i<argc;++i)
+            ++operations[argv[i]];
```

Comment:
increment on `bool` is deprecated in C++17. Please just set to `true. Multiple same options will be just ignored.